### PR TITLE
httpd add CMake build and CI

### PR DESCRIPTION
**Pull request by TomRoSystems**

### CODE REVIEW

File: instrumentation/httpd/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
+  "/usr/include/apache2"
+  "/usr/include/apr-1.0"
+)
+
+target_link_libraries(otel_httpd_module
+  PRIVATE
+  ${OPENTELEMETRY_CPP_LIBRARIES}
+  gRPC::grpc++
+#  Thrift::thrift
```
Comment (seemk) [inline instrumentation/httpd/CMakeLists.txt:38]: Thrift is searched for, but is not used?

File: instrumentation/httpd/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+set_target_properties(otel_httpd_module PROPERTIES
+  PREFIX ""
+)
+
+target_include_directories(otel_httpd_module
+  PRIVATE
+  ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
+  "/usr/include/apache2"
+  "/usr/include/apr-1.0"
```
Comment (seemk) [inline instrumentation/httpd/CMakeLists.txt:31]: I think Apache headers should be searched for to omit hardcoding of this search path.

E.g. it should be possible to do `-DCMAKE_PREFIX_PATH="/usr/local/customapachedir"` and CMake will pick up the include paths from it.

A simple example how to to write such a CMake script can be found at https://github.com/seemk/FastHumanDetection/blob/master/modules/FindKinectV2.cmake (unrelated project, just a simple example how a header file can be searched for)

File: instrumentation/httpd/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+find_package(CURL REQUIRED)
+find_package(Thrift REQUIRED)
+
+add_library(otel_httpd_module SHARED
+  src/otel/mod_otel.cpp
+  src/otel/opentelemetry.cpp
+)
+
+target_compile_options(otel_httpd_module
+  PRIVATE -Wall -Wextra -fpermissive
```
Comment (seemk) [inline instrumentation/httpd/CMakeLists.txt:18]: Imho we should not allow for `-fpermissive` and fix the underlying issues instead

File: instrumentation/httpd/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(opentelemetry-httpd)
+
+find_package(opentelemetry-cpp REQUIRED)
+find_package(Threads REQUIRED)
+find_package(protobuf REQUIRED)
+find_package(gRPC REQUIRED)
+find_package(CURL REQUIRED)
```
Comment (seemk) [inline instrumentation/httpd/CMakeLists.txt:9]: Is curl needed?

File: instrumentation/httpd/src/otel/opentelemetry.cpp [hunk 5]
```diff
@@ -30,17 +30,17 @@ namespace httpd_otel
...
+const opentelemetry::v0::nostd::string_view kAttrHTTPMethod             = "http.method";
+const opentelemetry::v0::nostd::string_view kAttrHTTPScheme             = "http.scheme";
+const opentelemetry::v0::nostd::string_view kAttrHTTPHost               = "http.host";
+const opentelemetry::v0::nostd::string_view kAttrHTTPTarget             = "http.target";
+const opentelemetry::v0::nostd::string_view kAttrHTTPUrl                = "http.url";
+const opentelemetry::v0::nostd::string_view kAttrHTTPFlavor             = "http.flavor";
+const opentelemetry::v0::nostd::string_view kAttrHTTPClientIP           = "http.client_ip";
+const opentelemetry::v0::nostd::string_view kAttrNETPeerIP              = "net.peer.ip";
+const opentelemetry::v0::nostd::string_view kAttrHTTPStatusCode         = "http.status_code";
+const opentelemetry::v0::nostd::string_view kAttrHTTPResponseContentLen = "http.response_content_length";
```
Comment (seemk) [inline instrumentation/httpd/src/otel/opentelemetry.cpp:43]: My hunch is that the explicit namespaces are not needed here and the issue is somewhere else, can you double check?

File: instrumentation/httpd/CMakeLists.txt (hunk 3 above)
Comment (TomRoSystems) [reply instrumentation/httpd/CMakeLists.txt]: Yup. Fixed with `reinterpret_cast`

File: instrumentation/httpd/src/otel/opentelemetry.cpp (hunk 5 above)
Comment (TomRoSystems) [reply instrumentation/httpd/src/otel/opentelemetry.cpp]: You're right. I've removed those.

File: instrumentation/httpd/CMakeLists.txt (hunk 4 above)
Comment (TomRoSystems) [reply instrumentation/httpd/CMakeLists.txt:9]: Yes without it I've got an error: `... links to target "CURL::libcurl" but the target was not found.`

File: instrumentation/httpd/CMakeLists.txt (hunk 1 above)
Comment (TomRoSystems) [reply instrumentation/httpd/CMakeLists.txt:38]: Yes, as at the moment I'm not using Jaeger exporter yet. And it compiles - you can check results as I've added CMake to continuous integration to GitHub actions.

File: instrumentation/httpd/src/otel/opentelemetry.h [hunk 6]
```diff
@@ -85,8 +86,8 @@ struct HttpdEndSpanAttributes
 // to be called explictly inside Destruct method
 struct ExtraRequestData
 {  // context which we pass
-  nostd::unique_ptr<opentelemetry::context::Token> token;
-  nostd::shared_ptr<opentelemetry::v0::trace::Span> span;
+  opentelemetry::v0::nostd::unique_ptr<opentelemetry::context::Token> token;
```
Comment (seemk) [inline instrumentation/httpd/src/otel/opentelemetry.h:89]: Is the `v0` prefix needed?

Comment (TomRoSystems) [reply instrumentation/httpd/src/otel/opentelemetry.h]: Prefix removed.

**Review by seemk — APPROVED** [review]

**Pull request merged**